main.go

```diff
@@ -1076,6 +1086,158 @@ func main() {
 		return Redirect(fmt.Sprintf("/users/%s/books/%s", user.Slug, vars["isbn"]))
 	}, loggedinMiddleware)
 
+	POST("/users/export/{user}", func(w Response, r Request) Output {
+		vars := VARS(r)
+
+		user, err := Q.UserBySlug(r.Context(), vars["user"])
+		if err != nil {
+			return NotFound
+		}
+
+		// export books data
+		books, err := Q.ExportBooksByUserId(r.Context(), user.ID);
+		if err != nil {
+			return InternalServerError(err)
+		}
+
+		booksFile, _ := json.MarshalIndent(books, "", "")
+		err = ioutil.WriteFile(BOOK_EXPORT_PATH, booksFile, 0644)
+		if err != nil {
+			return InternalServerError(err)
+		}
+		
+		buf := new(bytes.Buffer)
+		writer := zip.NewWriter(buf)
+
+		bookData, err := ioutil.ReadFile(BOOK_EXPORT_PATH)
+		if err != nil {
+			return InternalServerError(err)
+		}
+		
+		// Write books.json file into the zip file
+		f_book, err := writer.Create(BOOK_EXPORT_PATH)
+		if err != nil {
+			return InternalServerError(err)
+		}
+		_, err = f_book.Write([]byte(bookData))
```

Comment:
instead of writing `bookData` you can move the block that marshal books line 1103 here and write to directly to writer. so you won't need to write to a file then read it again. that way you can skip writing to the disk completely